app/src/main/java/fr/free/nrw/commons/filepicker/UploadableFile.java

```diff
@@ -124,13 +126,29 @@ private DateTimeWithSource getFileCreatedDateFromCP(Context context) {
     private DateTimeWithSource getDateTimeFromExif() {
         try {
             ExifInterface exif = new ExifInterface(file.getAbsolutePath());
-            @SuppressLint("RestrictedApi") Long dateTime = exif.getDateTime();
+            // TAG_DATETIME returns the last edited date, we need TAG_DATETIME_ORIGINAL for creation date
+            // See issue https://github.com/commons-app/apps-android-commons/issues/1971
+            String dateTimeSubString = exif.getAttribute(ExifInterface.TAG_DATETIME_ORIGINAL);
+
+            String year = dateTimeSubString.substring(0,4);
```

Comment:
[getAttribute](https://developer.android.com/reference/android/media/ExifInterface#getAttribute(java.lang.String)) may return null, so I think you will need to check that first.

(I would personally prefer checking the size than catching IndexOutOfBoundsException, but that's not a huge issue anyway.)

Reply:
[Here is a sample file without EXIF](https://user-images.githubusercontent.com/10674/181870174-14a8458f-dc69-4cc8-b225-197b10bd7234.jpg) - trying to upload this gave a NullPointerException when I tested.